server/src/main/java/org/elasticsearch/transport/TransportService.java

```diff
@@ -265,8 +266,9 @@ public void onFailure(Exception e) {
                     }
                     @Override
                     public void doRun() {
-                        // cf. ExceptionsHelper#isTransportStoppedForAction
-                        TransportException ex = new TransportException("transport stopped, action: " + holderToNotify.action());
+                        TransportException ex =
+                            new SendRequestTransportException(localNode, holderToNotify.action(), new NodeClosedException(localNode));
```

Comment:
Use `holderToNotify.connection().getNode()` as first parameter to `SendRequestTransportException`? I think this is supposed to be the node we're sending the request to.

Reply:
Thanks, fixed in c6c741f5251618cefaa7a2608533a226d4df9759